lib/src/linting/rules/has-request-payload.ts

```diff
@@ -0,0 +1,57 @@
+import { Locatable } from "lib/src/models/locatable";
+import { EndpointNode } from "lib/src/models/nodes";
+import { complement } from "ramda";
+import { LintingRule } from "../rule";
+
+/**
+ * Checks that the request payload is defined if and only if the method is POST/PUT/PATCH.
+ */
+export const hasRequestPayload: LintingRule = contract => {
+  return [
+    ...mutationEndpointsHaveRequestPayload(contract),
+    ...nonMutationEndpointsDoNotHaveRequestPayload(contract)
+  ];
+};
+
+const mutationEndpointsHaveRequestPayload: LintingRule = contract => {
+  return contract.endpoints
+    .filter(isMutationEndpoint)
+    .filter(complement(endpointHasRequestPayload))
+    .map(endpoint => ({
+      message: `${
+        endpoint.value.name.value
+      } should have a request payload as its method is ${
+        endpoint.value.method.value
+      }`,
+      source: endpoint
+    }));
+};
+
+const nonMutationEndpointsDoNotHaveRequestPayload: LintingRule = contract => {
+  return contract.endpoints
+    .filter(complement(isMutationEndpoint))
+    .filter(endpointHasRequestPayload)
+    .map(endpoint => ({
+      message: `${
+        endpoint.value.name.value
+      } should not have a request payload as its method is ${
+        endpoint.value.method.value
+      }`,
+      source: endpoint
+    }));
+};
+
+function isMutationEndpoint(endpoint: Locatable<EndpointNode>) {
+  switch (endpoint.value.method.value) {
```

Comment:
can be simplified to something like `MUTATION_METHODS = ["POST", "PUT", "PATCH"]` `MUTATION_METHODS.includes(endpoint.value.method.value)`

Reply:
I like it! But it's not as efficient and I'd argue a bit easier to break (you'd need `MUTATION_METHODS: HttpMethod[] = ["POST", "PUT", "PATCH"]` to make it safer, and that's less readable). I'll keep the switch syntax for now.